tests/test_time_series.py

```diff
@@ -0,0 +1,187 @@
+)
+
+
+class TestTimeSeriesChecks(TestCase):
+    def setUp(self):
+        self.tempdir = Path(mkdtemp())
+        self.nwbfile = pynwb.NWBFile(
+            session_description="Testing inspector.",
+            identifier=str(uuid4()),
+            session_start_time=datetime.now().astimezone(),
+        )
+        self.nwbfile_path = str(self.tempdir / "testing.nwb")
+
+    def tearDown(self):
+        rmtree(self.tempdir)
+
+    def test_check_dataset_compression(self):
+        time_series = pynwb.TimeSeries(
+            name="test_time_series",
+            unit="test_unit",
+            data=np.zeros(shape=int(3e6 / np.dtype("float").itemsize)),
+            rate=1.0,
+        )
+
+        output_message_1 = check_dataset_compression(time_series=time_series)
+        self.assertEqual(first=output_message_1, second=None)
+
+        self.nwbfile.add_acquisition(time_series)
+        output_message_2 = check_dataset_compression(
+            time_series=self.nwbfile.acquisition["test_time_series"]
+        )
+        self.assertEqual(first=output_message_2, second=None)
+
+        with pynwb.NWBHDF5IO(path=self.nwbfile_path, mode="w") as io:
+            io.write(self.nwbfile)
+        with pynwb.NWBHDF5IO(path=self.nwbfile_path, mode="r") as io:
+            written_nwbfile = io.read()
+            output_message_3 = check_dataset_compression(
+                time_series=written_nwbfile.acquisition["test_time_series"]
+            )
+            self.assertEqual(
+                first=output_message_3,
+                second="Consider enabling compression when writing a large dataset.",
+            )
+
+    def test_check_regular_timestamps(self):
+        time_series = pynwb.TimeSeries(
+            name="test_time_series",
+            unit="test_units",
+            data=np.zeros(shape=3),
+            timestamps=[1.2, 3.2, 5.2],
+        )
+        true_message = (
+            f"The {type(time_series).__name__} '{time_series.name}' has a constant sampling rate. "
+            f"Consider specifying starting_time={time_series.timestamps[0]} "
+            f"and rate={time_series.timestamps[1] - time_series.timestamps[0]} instead of timestamps."
+        )
+
+        output_message_1 = check_regular_timestamps(time_series=time_series)
+        self.assertEqual(first=output_message_1, second=true_message)
+
+        self.nwbfile.add_acquisition(time_series)
+        output_message_2 = check_regular_timestamps(
+            time_series=self.nwbfile.acquisition["test_time_series"]
+        )
+        self.assertEqual(first=output_message_2, second=true_message)
+
+        with pynwb.NWBHDF5IO(path=self.nwbfile_path, mode="w") as io:
+            io.write(self.nwbfile)
+        with pynwb.NWBHDF5IO(path=self.nwbfile_path, mode="r") as io:
+            written_nwbfile = io.read()
+            output_message_3 = check_regular_timestamps(
+                time_series=written_nwbfile.acquisition["test_time_series"]
+            )
```

Comment:
It looks like this logic is repeated a few times. Is there a good way to refactor this code to reduce this code duplication?

Reply:
That would be easily done, yes, pending modularity decision.